app/src/main/java/com/example/poste/models/Folder.java

```diff
@@ -60,6 +69,34 @@ public String getUserPermission() {
         return userPermission;
     }
 
+    /**
+     * Adds a new post the given folder
+     *
+     * @param folder folder to add the new post to
+     */
+    public Post createNewPost(Folder folder) {
+        Post.Builder pb = new Post.Builder();
+        pb.setTitle("Untitled");
+        Post newPost = pb.build();
+        MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+        Call<ResponseBody> call = apiService.createPost(new CreatePost(newPost.getTitle(), newPost.getDescription(), newPost.getUrl(), folder.getTitle()));
+        call.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                posts.add(newPost);
+            }
+            @Override
+            public void onFailure(Call<ResponseBody> call, Throwable t) {
+            }
+        });
+        if (posts.contains(newPost)) {
```

Comment:
This can potentially throw a NullPointerException elsewhere; it is called in FolderViewActivity (and perhaps elsewhere?) and assigned to a variable, which is then set as SelectedPost, which is used elsewhere, etc.

Reply:
Method now returns void. no longer an issue